src/main/java/org/interledger/psk/PskMessage.java

```diff
@@ -0,0 +1,410 @@
+package org.interledger.psk;
+
+import org.interledger.InterledgerRuntimeException;
+
+import java.time.OffsetDateTime;
+import java.time.temporal.TemporalAmount;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+/**
+ * This interface represents the message format as defined in the Pre-Shared Key Transport Protocol.
+ */
+public interface PskMessage {
+
+  String STATUS_LINE = "PSK/1.0";
+
+  /**
+   * Get the default builder.
+   *
+   * @return a {@link Builder} instance.
+   */
+  static Builder builder() {
+    return new Builder();
+  }
+
+  /**
+   * Returns a list of all public headers in the message. Note that all parties may view the public
+   * headers.
+   */
+  List<Header> getPublicHeaders();
+
+  /**
+   * Returns the all the <b>public</b> headers with the specified name.
+   *
+   * @param headerName The name of the header(s) to return
+   *
+   * @return the <b>public</b> headers whose name matches the parameter, or an empty list.
+   */
+  List<Header> getPublicHeaders(String headerName);
+
+  /**
+   * Returns a list of all <b>private</b> headers in the message, provided that either the message
+   * was <b>not</b> encrypted, or the private portion of the message has been decrypted by the
+   * receiver.
+   */
```

Comment:
Minor nit, but we should duplicate the Javadoc clarification from `getPrivateHeaders(headerName)` to overtly state the returned list is empty in certain circumstances:

```
 @return the <b>private</b> headers for this PskMessage, or an empty list.
```